src/test/java/com/scottlogic/filters/ContentFilterTest.java

```diff
@@ -0,0 +1,89 @@
+
+        return new UserPost("Joe Bloggs",
+                OffsetDateTime.of(2020, 1, 3, 7, 12, 3, 0, ZoneOffset.UTC),
+                content, 2);
+    }
+
+    @Test
+    public void filter_validKeywordSearch_correctListReturned() {
+        ContentFilter filterByKeyword = new ContentFilter("example");
+        UserPost userPost1 = createUserPost("An example of a post \nwith lines breaks.");
+        UserPost userPost2 = createUserPost("Another example post.");
+        UserPost userPost3 = createUserPost("Hello World!");
+
+        List<UserPost> userPosts = Arrays.asList(userPost1, userPost2, userPost3);
+        List<UserPost> actualResult = filterByKeyword.filter(userPosts);
+        List<UserPost> expectedResult = Arrays.asList(userPost1, userPost2);
+        Assert.assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    public void filter_nullSearch_emptyReturned() {
+        ContentFilter filterByKeyword = new ContentFilter(null);
+        UserPost userPost1 = createUserPost("An example of a post \nwith lines breaks.");
+        UserPost userPost2 = createUserPost("Another example post.");
+
+        List<UserPost> userPosts = Arrays.asList(userPost1, userPost2);
+        List<UserPost> actualResult = filterByKeyword.filter(userPosts);
+        Assert.assertTrue(actualResult.isEmpty());
+    }
+
+    @Test
+    public void filter_nullContent_correctListReturned() {
+        ContentFilter filterByKeyword = new ContentFilter("example");
+        UserPost userPost1 = createUserPost("An example of a post \nwith lines breaks.");
+        UserPost userPost2 = createUserPost(null);
+        UserPost userPost3 = createUserPost("Another example post.");
+
+        List<UserPost> userPosts = Arrays.asList(userPost1, userPost2, userPost3);
+        List<UserPost> actualResult = filterByKeyword.filter(userPosts);
+        List<UserPost> expectedResult = Arrays.asList(userPost1, userPost3);
+        Assert.assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    public void filter_nullPostInList_nullIgnored() {
+        ContentFilter filterByKeyword = new ContentFilter("post");
+        UserPost userPost1 = createUserPost("An example of a post \nwith lines breaks.");
+        UserPost userPost2 = createUserPost("Another example post.");
+        UserPost nullPost = null;
+        List<UserPost> userPosts = Arrays.asList(userPost1, userPost2, nullPost);
+        List<UserPost> actualResult = filterByKeyword.filter(userPosts);
+        List<UserPost> expectedResult = Arrays.asList(userPost1, userPost2);
+        Assert.assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    public void filter_emptyList_emptyListReturned() {
+        ContentFilter filterByKeyword = new ContentFilter("post");
+        List<UserPost> userPosts = new ArrayList<>();
+        List<UserPost> actualResult = filterByKeyword.filter(userPosts);
+        Assert.assertTrue(actualResult.isEmpty());
+    }
```

Comment:
We're missing a test for a null input list here. It would also be good to see one where the search term is not in the input list.